sympy/stats/crv_types.py

```diff
@@ -2195,6 +2196,110 @@ def Normal(name, mean, std):
             MultivariateNormalDistribution, name, mean, std)
     return rv(name, NormalDistribution, (mean, std))
 
+
+#-------------------------------------------------------------------------------
+# Inverse Normal distribution ----------------------------------------------------------
+
+
+class NormalInverseDistribution(SingleContinuousDistribution):
+    _argnames = ('mean', 'shape')
+
+    @property
+    def set(self):
+        return Interval(0, oo)
+
+    @staticmethod
+    def check(mean, shape):
+        _value_check(shape > 0, "Shape parameter must be positive")
+        _value_check(mean > 0, "Mean must be positive")
+
+    def pdf(self, x):
+        return exp(-self.shape*(x - self.mean)**2 / (2*x*self.mean**2)) * sqrt(self.shape/((2*pi*x**3)))
+
+    def sample(self):
+        scipy = import_module('scipy')
+        if scipy:
+            from scipy.stats import invgauss
+            return invgauss.rvs(float(self.mean/self.shape), 0, float(self.shape))
+        else:
+            raise NotImplementedError('Sampling the Inverse Normal Distribution requires Scipy.')
+
+    def _standardNormal_cdf(self, x):
+        return erf(sqrt(2)*(x - S.One)/2)/2 + S.Half
+
+    def _cdf(self, x):
+        mean, shape = self.mean, self.shape
+        return self._standardNormal_cdf(sqrt(shape/x) * ((x/mean - S.One))) + exp(2*shape/mean)*\
```

Comment:
Is it possible to write this as a combination of terms? I am saying so, because in the case of `GumbelDistribution` the PDF got wrong and the mistake was undiscovered for a long time due to a confusing representation of PDF. It will be great if you do the splitting.

Reply:
Done. Thanks for the suggestion!